### ISSUE TITLE
Incrimental Reading Feature

### ISSUE THREAD

**Issue by Nick3C** [batterseapower/pinyin-toolkit#93]
The way we are now using dictionary for lookups means it is not a strech to build-in an incrimental reader. I am extremly keen to do this to support natural learning methods. If you have ever used sites that support such methods such as LingQ.com (take a quick look) then you will be familiar with the type of reader I am talking about. I believe supermemo had such a feature but I have never used the software.

The idea combines naturally with Anki, read using the IR, then review using Anki. If we combine the two Anki could easily become the primary chinese lanugaeg tool on the open-source web. The project is not massive, but it is not trivial either so probably needs to be implemented over time. I would like to consider it high priority. I do not think it is wise to develop a university i.reader (because Chinese differs quite markedly from other languages, that is why LingQ handles Chinese badly, for example).

The basic idea is a web display/text editor with a heirachical tree on the left side. The tree displays readable files and when selected they appear in the main text display/browser. It is preferable that they are not editable and they appear as html. The files should be chiense texts. The idea is that you read through the text, selecting words that you don't know and adding cards for them as you go along. We could do some tight integration with Anki card generation for this. Doing this marks the text in some way (probably background grey) to show you have read it [using css class tags).

The benefits of this approach are massive. If you only had time to do a few lines then that is fine. You could always come back later and resume from where you left off. You are no longer restricted to short texts. Users with slow-reading speeds can thus work on longer texts, and their speed will increase markedly as they work. At the same time you are building a library of cards you need to know (at your reading level) in Anki. At the same time you are also building a library of processed texts that you can review in the future to refresh your memory.

It would also have all sort of related benefits such as being able to specify the destination file for the Firefox Plugin Chinese pera-kun into this directory. The result would be extremly easy importing of those words into Anki.

I am rather busy at the moment so not sure when I will be able to start looking at this. However it is imporant so I wanted to note it here. Also hope to discuss at some point (and do check out LingQ if you get a chance).

Design notes (incomplete):
- on clicking toolbar or menu open incrimental reading window
- window:
  - window the size of current preferences window
  - left-hand side a list of files in the "Readings" directory (respects heirachy) for extensions: .txt, .u8, .htm, .html, .wen., 
  - bottom of left-side has "add url", "open Readings Directory","refresh", and "Save" buttons
  - right-hand side large web browser window
  - top, controls menu:
    > generate card [mark selected text as read [css class="status-read"], pop-up add card window and paste current text into selected field, update readings, close window on pressing 'add' and refocus on IR Window]
    > mark as read [umark as read using css to create a block background effect, probably color in dark grey]
    > mark as unread [remove the status-read css tag to uncolor the text]
- text saves on window close or if different file selected in the tree
- add menu option to "add a URL" which will download an html page and save in folder
- add tools menu: Intrimental reading
- add toolbar: incrimental reading
- add preferes:
  - IR enabled/disabled

**Comment by Nick3C** [comment]
Did you have any thoughts on this one Max?

**Comment by batterseapower** [comment]
I did look at the LingQ website. It does sound neat, but it's a big chunk of work - requires quite a lot of well-done UI. I'm trying to focus on the more fundamental issues plaguing the TK at the moment, such as speed (or lack of it).

**Comment by Nick3C** [comment]
Yes, agreed. It should be good but it is fairly substantial.and can wait.
